### PR TITLE
Derive serde's Serialize and Deserialize for structs in ibc-proto::ibc::core and ibc_proto::ibc::applications 

### PR DESCRIPTION
**Pull request by Wizdave97**
Closes: #1988 

## Description

Implements Serialize and Deserialize for ibc core and applications types.

______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests: integration (for Hermes) or unit/mock tests (for modules). 
- [x] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: proto-compiler/src/cmd/compile.rs [hunk 1]
```diff
@@ -89,25 +89,55 @@ impl CompileCmd {
...
+            "#[derive(Eq, PartialOrd, Ord)]";
+        let attrs_eq = "#[derive(Eq)]";
         let attrs_serde_default = "#[serde(default)]";
         let attrs_serde_base64 = r#"#[serde(with = "crate::base64")]"#;
         let attrs_jsonschema_str =
             r#"#[cfg_attr(feature = "json-schema", schemars(with = "String"))]"#;
 
         let compilation = tonic_build::configure()
             .build_client(true)
+            .compile_well_known_types(true)
```
Comment (romac) [inline proto-compiler/src/cmd/compile.rs:105]: @Wizdave97 Do you need this for your use case or can we keep this change for a separate PR to make it easier to review?

Comment (Wizdave97) [reply proto-compiler/src/cmd/compile.rs:105]: Yeah, the `Any` type is used in some of those structs so it also needs the serialize and deserialize implementation

Comment (romac) [reply proto-compiler/src/cmd/compile.rs:105]: Can you please go into a bit more details? Is the issue that prost_types::Any does not implement (De)Serialize?

Comment (Wizdave97) [reply proto-compiler/src/cmd/compile.rs:105]: Yeah prost_types::Any does not implement (De)Serialize and I felt making a longer time to get it merged

Comment (Wizdave97) [reply proto-compiler/src/cmd/compile.rs:105]: Also seems like their release cycles are longer, the last release was October last year.

**Comment by romac** [comment]
@Wizdave97 Can you please allow "edits by maintainers" for this PR, cf. https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Comment by romac** [comment]
@Wizdave97 Can you please [allow edits by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for this PR?

**Comment by Wizdave97** [comment]
> @Wizdave97 Can you please [allow edits by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for this PR?

It's already allowed

**Comment by romac** [comment]
Somehow I am not able to push to the branch, but that's okay let's just move this to #2005.
